pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -344,9 +344,10 @@ func (r *ReconcileDeadmansSnitchIntegration) dmsAddFinalizer(dmsi *deadmanssnitc
 func (r *ReconcileDeadmansSnitchIntegration) createSnitch(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, cd *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
 	logger := log.WithValues("DeadMansSnitchIntegration.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
 	snitchName := utils.DmsSnitchName(cd.Spec.ClusterName, cd.Spec.BaseDomain, dmsi.Spec.SnitchNamePostFix)
+	clusterID := cd.Spec.ClusterMetadata.ClusterID
```

Comment:
What if `cd.Spec.ClusterMetadata` is nil?

Reply:
Good point. Previously (before my PRs) this has never been checked. https://github.com/openshift/deadmanssnitch-operator/blob/9a1a9278f8b40339092d937046935737705d7964/pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go#L364 

If we are checking for `cd.Spec.ClusterMetadata` for nil here, should we also be checking if `cd.Spec` and/or `cd`?